src/java/org/apache/cassandra/cql3/functions/BytesConversionFcts.java

```diff
@@ -18,92 +18,104 @@
 package org.apache.cassandra.cql3.functions;
 
 import java.nio.ByteBuffer;
-import java.util.HashSet;
 import java.util.List;
-import java.util.Set;
 
 import org.apache.cassandra.cql3.CQL3Type;
-import org.apache.cassandra.db.marshal.AbstractType;
 import org.apache.cassandra.db.marshal.BytesType;
-import org.apache.cassandra.db.marshal.UTF8Type;
-import org.apache.cassandra.transport.ProtocolVersion;
-import org.apache.cassandra.utils.ByteBufferUtil;
 import org.apache.cassandra.exceptions.InvalidRequestException;
 import org.apache.cassandra.serializers.MarshalException;
+import org.apache.cassandra.transport.ProtocolVersion;
+import org.apache.cassandra.utils.ByteBufferUtil;
 
 public abstract class BytesConversionFcts
 {
     public static void addFunctionsTo(NativeFunctions functions)
     {
-        // because text and varchar ends up being synonymous, our automatic makeToBlobFunction doesn't work
```

Comment:
This has been removed in the new code?

Reply:
I have changed the function builders to receive the CQL type, instead of the internal one. That makes the special-casing for {{varchar}} and the {{Set}} unnecessary.